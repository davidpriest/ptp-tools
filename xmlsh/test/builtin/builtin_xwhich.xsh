# Test of xwhich 

foo()
{ echo bar ; }

# test of builtin
xwhich echo 

# test of internal
xwhich xfile

# test of function
xwhich foo 

# test of script
F=$<(xwhich builtin_xwhich.xsh)
echo xwhich is <[$F/xwhich/command/@name/string()]>

# test of external
# PATH MAY VARY - only output name
F=$<(xwhich rm)
echo rm is <[$F/xwhich/command/@name/string()]>
