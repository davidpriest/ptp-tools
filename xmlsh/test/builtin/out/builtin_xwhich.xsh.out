<xwhich>
   <command name="echo" type="builtin"/>
</xwhich>
<xwhich>
   <command name="xfile" type="internal" module="xmlsh"/>
</xwhich>
<xwhich>
   <command name="foo" type="function"/>
</xwhich>
xwhich is builtin_xwhich.xsh
rm is rm
